packages/devtools_app/lib/src/shared/utils.dart

```diff
@@ -159,3 +159,19 @@ mixin ProvidedControllerMixin<T, V extends StatefulWidget> on State<V> {
     return true;
   }
 }
+
+String? _cachedIsolateId;
+String? _cachedRootPackage;
```

Comment:
Would these cached values be frequently garbage collected since they wouldn't have a permanent instance that they live in?

also how often do isolateId's conflict between applications. Would we ever run into a case where we grab the rootPackage of another application since their isolateIds were identical?

Reply:
Another thing I'm thinking of, should we be persisting these in a dictionary incase we reconnect to the application and want to grab this again?

Reply:
Globals are never garbage collected.
I am ok to re-query the package in case of reconnection, as it is rare case and is not worth code complexity IMHO.
Moved it to serviceManager ant protected from conflict. Thank you.